egs/wsj/s5/steps/nnet3/blstm/make_configs.py

```diff
@@ -0,0 +1,235 @@
+    f.close()
+
+    config_lines = {'components':[], 'component-nodes':[]}
+
+    config_files={}
+    prev_layer_output = nodes.AddInputLayer(config_lines, args.feat_dim, splice_indexes[0], args.ivector_dim)
+
+    # Add the init config lines for estimating the preconditioning matrices
+    init_config_lines = copy.deepcopy(config_lines)
+    init_config_lines['components'].insert(0, '# Config file for initializing neural network prior to')
+    init_config_lines['components'].insert(0, '# preconditioning matrix computation')
+    nodes.AddOutputNode(init_config_lines, prev_layer_output)
+    config_files[args.config_dir + '/init.config'] = init_config_lines
+
+    prev_layer_output = nodes.AddLdaLayer(config_lines, "L0", prev_layer_output, args.config_dir + '/lda.mat')
+
+    for i in range(args.num_lstm_layers):
+	if len(lstm_delay[i]) == 2: # BLSTM layer case, add both forward and backward
+            prev_layer_output1 = nodes.AddLstmLayer(config_lines, "Lstm{0}f".format(i+1), prev_layer_output, args.cell_dim,
+                                             args.recurrent_projection_dim, args.non_recurrent_projection_dim,
+                                             args.clipping_threshold, args.norm_based_clipping,
+                                             args.ng_per_element_scale_options, args.ng_affine_options,
+                                             lstm_delay = lstm_delay[i][0])
+            prev_layer_output2 = nodes.AddLstmLayer(config_lines, "Lstm{0}b".format(i+1), prev_layer_output, args.cell_dim,
+                                             args.recurrent_projection_dim, args.non_recurrent_projection_dim,
+                                             args.clipping_threshold, args.norm_based_clipping,
+                                             args.ng_per_element_scale_options, args.ng_affine_options,
+                                             lstm_delay = lstm_delay[i][1])
+            prev_layer_output['descriptor'] = 'Append({0}, {1})'.format(prev_layer_output1['descriptor'], prev_layer_output2['descriptor'])
+	    prev_layer_output['dimension'] = prev_layer_output1['dimension'] + prev_layer_output2['dimension']
+	else: # LSTM layer case
+	    prev_layer_output = nodes.AddLstmLayer(config_lines, "Lstm{0}".format(i+1), prev_layer_output, args.cell_dim,
+			                    args.recurrent_projection_dim, args.non_recurrent_projection_dim,
+					    args.clipping_threshold, args.norm_based_clipping,
+					    args.ng_per_element_scale_options, args.ng_affine_options,
+					    lstm_delay = lstm_delay[i][0])
+        # make the intermediate config file for layerwise discriminative
+        # training
+        nodes.AddFinalLayer(config_lines, prev_layer_output, args.num_targets, args.ng_affine_options, args.label_delay)
+        config_files['{0}/layer{1}.config'.format(args.config_dir, i+1)] = config_lines
+        config_lines = {'components':[], 'component-nodes':[]}
+	if len(lstm_delay[i]) == 2:
```

Comment:
I dont follow the code at line 202 and 205 you already set prev_layer_output. Now you are overwriting the prev_layer_output. I don't any recursive definitions involution prev_layer_output so I don't see why you would have the pathological case ?